python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,599 @@
+        return result
+
+    def __repr__(self):
+        return (
+            super().__repr__() + "\noffsets: " + self.offsets.__repr__() + "\n"
+        )
+
+    def copy(self, deep=True):
+        result = super().copy(deep)
+        result.offsets = self.offsets.copy(deep)
+        return result
+
+
+class GpuLineArray(GpuOffsetArray):
+    def __init__(self):
+        super().__init__()
+
+    def __getitem__(self, index):
+        result = super().__getitem__(index)
+        return result
+
+    def __repr__(self):
+        return (
+            super().__repr__() + "\nmlines: " + self.mlines.__repr__() + "\n"
+        )
+
+    def copy(self, deep=True):
+        result = GpuLineArray()
+        base = super().copy(deep)
+        result.xy = base.xy
+        result.z = base.z
+        result.offsets = base.offsets
+        if hasattr(self, "mlines"):
+            result.mlines = self.mlines.copy()
+        return result
+
+
+class GpuMultiPointArray(GpuOffsetArray):
+    def __init__(self):
+        super().__init__()
+
+    def copy(self, deep=True):
+        result = GpuMultiPointArray()
+        base = super().copy(deep)
+        result.xy = base.xy
+        result.z = base.z
+        result.offsets = base.offsets
+        return result
+
+
+class GpuPolygonArray(GpuOffsetArray):
+    def __init__(self):
+        super().__init__()
+        self.polys = None
+        # GpuPolygonArray uses the offsets buffer for rings!
+        self.mpolys = None
+
+    @property
+    def rings(self):
+        return self.offsets
+
+    @rings.setter
+    def rings(self, rings):
+        self.offsets = rings
+
+    def __repr__(self):
+        result = ""
+        result += "xy:\n" + self.xy.__repr__() + "\n"
+        result += "polys:\n" + self.polys.__repr__() + "\n"
+        result += "rings:\n" + self.rings.__repr__() + "\n"
+        result += "mpolys:\n" + self.mpolys.__repr__() + "\n"
+        return result
+
+    def copy(self, deep=True):
+        result = super().copy(deep)
+        result.polys = self.polys.copy(deep)
+        result.rings = self.rings.copy(deep)
+        result.mpolys = self.mpolys.copy(deep)
+        return result
+
+
+class gpuGeometry:
+    def __init__(self, source, index):
+        self.source = source
+        self.index = index
+
+    def __repr__(self):
+        return self.source.__repr__()
+
+
+class gpuPoint(gpuGeometry):
+    def to_shapely(self):
+        item_type = self.source.types[self.index]
+        index = (
+            pd.Series(self.source.types[0 : self.index], dtype="object")
+            == item_type
+        ).sum()
+        return Point(self.source._points[index].reset_index(drop=True))
+
+    def __iter__(self):
+        self._index = 0
+        self._shapely = self.to_shapely()
+        return self
+
+    def __next__(self):
+        if self._index >= 1:
+            raise StopIteration
+        result = self._shapely
+        self._index = self._index + 1
+        return result
```

Comment:
Why are we iterating over GPU backed objects? Also, it looks like if someone uses `__next__`, `self._shapely` needs to be updated?

Reply:
I think this is crust, and I'm removing it.